From d25aca44fcc377b7d370603197d87c2faab4df26 Mon Sep 17 00:00:00 2001
From: Ariel Aviad <aaviad@habana.ai>
Date: Tue, 31 Dec 2024 12:02:00 +0200
Subject: [PATCH] [SW-204890] hl-smi: info: print power limit constraints

Print information about possible values of power
management limits of a device

Move MV_TO_VOLT define to be global

Change-Id: Iba0226ba346cf453b0edfea778c54aaac2e43b93
Signed-off-by: Ariel Aviad <aaviad@habana.ai>
---

diff --git a/app/hl-smi/info.cpp b/app/hl-smi/info.cpp
index 0b97eec..5ad2249 100644
--- a/app/hl-smi/info.cpp
+++ b/app/hl-smi/info.cpp
@@ -566,8 +566,8 @@
 static void hl_smi_aip_log_power(struct hl_smi_aip_info *aip)
 {
 	hlml_return_t ret;
-	char info_field[VERSION_MAX_LEN] = {0};
 	unsigned int result;
+	uint32_t power_min, power_max;
 	hlml_enable_state_t pm_mng_mode;
 	char const *pm_mng_mode_state[] = {
 		[HLML_FEATURE_DISABLED] = "N/A",
@@ -594,12 +594,14 @@
 	else
 		HL_SMI_LOG_INFO("\t\tPower Max\t\t: {} W", result/1000);
 
-	if (aip->power_limit[0])
-		snprintf(info_field, VERSION_MAX_LEN, "%s W", aip->power_limit);
-	else
-		snprintf(info_field, VERSION_MAX_LEN, "%s", HL_NA_STR);
-
-	HL_SMI_LOG_INFO("\t\tPower Limit\t\t: {}", info_field);
+	ret = hlml_device_get_power_management_limit_constraints(aip, &power_min, &power_max);
+	if (ret != HLML_SUCCESS) {
+		HL_SMI_LOG_INFO("\t\tPower Limit Min\t\t: {}", HL_NA_STR);
+		HL_SMI_LOG_INFO("\t\tPower Limit Max\t\t: {}", HL_NA_STR);
+	} else {
+		HL_SMI_LOG_INFO("\t\tPower Limit Min\t\t: {} W", MV_TO_VOLT(power_min));
+		HL_SMI_LOG_INFO("\t\tPower Limit Max\t\t: {} W", MV_TO_VOLT(power_max));
+	}
 
 	if (aip->psu_sec_info) {
 		HL_SMI_LOG_INFO("\tPower Readings {}", aip->psu_sec_info->psu_sec_name);
diff --git a/include/hl-smi.h b/include/hl-smi.h
index b43610b..08609d3 100644
--- a/include/hl-smi.h
+++ b/include/hl-smi.h
@@ -96,6 +96,9 @@
 #define NAME_LENGTH 		8
 #define NAME_MAX_LENGTH		(2 * NAME_LENGTH)
 
+/* Power Defines*/
+#define MV_TO_VOLT(mv)	((mv) / 1000.0)
+
 /* Flags for hlml lib init */
 #define HL_SMI_APP		(1 << 0)
 #define HLML_ERROR_LOG		(1 << 1)
diff --git a/src/sensors.c b/src/sensors.c
index 8e8e4af..750b619 100644
--- a/src/sensors.c
+++ b/src/sensors.c
@@ -16,7 +16,6 @@
 #include "log.h"
 
 #define SENSORS_CHIP_NAME_SIZE 200
-#define MV_TO_VOLT(mv)	((mv) / 1000.0)
 
 enum sensors_data_type {
 	SENSORS_DATA_TYPE_UINT,
